### PR TITLE
[Windows] Fix system theme integration.

### PR DESCRIPTION
**Pull request by moko256**
The #35019 didn't merge [SettingsPluginWin32's constructor and destructor](https://github.com/flutter/engine/blob/c4f583d1baaadaa3a65502382c4e3745d5e937d4/shell/platform/windows/settings_plugin_win32.cc#L24-L34). This PR adds them and fix SettingsPlugin to re-enable it to watch the theme changes.

This PR is part of flutter/flutter#110700.

No changes in the [flutter/tests] repo.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/windows/settings_plugin.cc [hunk 1]
```diff
@@ -34,9 +34,18 @@ SettingsPlugin::SettingsPlugin(BinaryMessenger* messenger,
           messenger,
           kChannelName,
           &JsonMessageCodec::GetInstance())),
-      task_runner_(task_runner) {}
+      task_runner_(task_runner) {
+  RegOpenKeyEx(HKEY_CURRENT_USER, kGetPreferredBrightnessRegKey,
+               RRF_RT_REG_DWORD, KEY_NOTIFY, &preferred_brightness_reg_hkey_);
+  RegOpenKeyEx(HKEY_CURRENT_USER, kGetTextScaleFactorRegKey, RRF_RT_REG_DWORD,
+               KEY_NOTIFY, &text_scale_factor_reg_hkey_);
```
Comment (loic-sharma) [inline shell/platform/windows/settings_plugin.cc:41]: Could we place these in `WatchPreferredBrightnessChanged` and `WatchTextScaleFactorChanged`? What are the benefits of putting this here?

Comment (moko256) [reply shell/platform/windows/settings_plugin.cc]: That's nice. I move them.

Comment (moko256) [reply shell/platform/windows/settings_plugin.cc]: done

File: shell/platform/windows/settings_plugin.cc [hunk 2]
```diff
@@ -55,6 +57,12 @@ void SettingsPlugin::SendSettings() {
 }
 
 void SettingsPlugin::StartWatching() {
+  RegOpenKeyEx(HKEY_CURRENT_USER, kGetPreferredBrightnessRegKey,
+               RRF_RT_REG_DWORD, KEY_NOTIFY, &preferred_brightness_reg_hkey_);
+  RegOpenKeyEx(HKEY_CURRENT_USER, kGetTextScaleFactorRegKey, RRF_RT_REG_DWORD,
+               KEY_NOTIFY, &text_scale_factor_reg_hkey_);
```
Comment (loic-sharma) [inline shell/platform/windows/settings_plugin.cc:63]: Could we make `preferred_brightness_reg_hkey_` and `text_scale_factor_reg_hkey_` protected and then test that these are non-null after calling `StartWatching`?

**Review by loic-sharma — APPROVED** [review]
LGTM once `preferred_brightness_reg_hkey_` and `text_scale_factor_reg_hkey_` are asserted to be non-null

**Review by yaakovschectman — APPROVED** [review]

File: shell/platform/windows/settings_plugin.cc (hunk 2 above)
Comment (moko256) [reply shell/platform/windows/settings_plugin.cc:63]: After the lines, those variables are checked whether they aren't null, and it affects whether WatchPreferredBrightnessChanged and WatchTextScaleFactorChanged is called.
Without this fix, the new test which checks they are called will fail.

**Review by cbracken — APPROVED** [review]
LGTM stamp from a Japanese personal seal

File: shell/platform/windows/settings_plugin.cc (hunk 2 above)
Comment (loic-sharma) [reply shell/platform/windows/settings_plugin.cc:63]: Ah gotcha, perfect!

**Pull request merged**